client.go

```diff
@@ -84,47 +83,38 @@ func (codec *clientCodec) ReadResponseHeader(response *rpc.Response) (err error)
 
 	codec.mutex.Lock()
 	httpResponse := codec.responses[seq]
+	delete(codec.responses, seq)
 	codec.mutex.Unlock()
 
 	if httpResponse.StatusCode < 200 || httpResponse.StatusCode >= 300 {
-		return fmt.Errorf("request error: bad status code - %d", httpResponse.StatusCode)
+		response.Error = fmt.Sprintf("request error: bad status code - %d", httpResponse.StatusCode)
 	}
 
-	respData, err := ioutil.ReadAll(httpResponse.Body)
-
+	defer httpResponse.Body.Close()
```

Comment:
We can close response body immideately atfer calling `ioutil.ReadAll`. I don't see much benefit of using `defer` here.